Src/CSharpier/SyntaxPrinter/RightHandSide.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using CSharpier.DocTypes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CSharpier.SyntaxPrinter
+{
+    public static class RightHandSide
+    {
+        public static Doc Print(ExpressionSyntax node)
+        {
+            var formatMode = node
+                is AnonymousObjectCreationExpressionSyntax
+                    or AnonymousMethodExpressionSyntax
+                    or InitializerExpressionSyntax
+                    or InvocationExpressionSyntax
+                    or ConditionalExpressionSyntax
+                    or ObjectCreationExpressionSyntax
+                    or SwitchExpressionSyntax
+                    or LambdaExpressionSyntax
+                    or AwaitExpressionSyntax
+                    or WithExpressionSyntax
+                ? FormatMode.NoIndent
```

Comment:
This looks like a nested ternary if on type matches. Usually, a [switch statement](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#multiple-inputs) looks better.

Not sure how good it will look with a dozen `or`s. But if it doesn't look good you can probably split them into multiple arms of the switch.